### ISSUE TITLE
Add TTL field to note

### ISSUE THREAD

**Issue by SISheogorath** [codimd/server#530]
I would like to see a TTL field in the public API. The TTL is used to indicate when to expire/delete a note. This allows anonymous notes to become more useful and get out of the category "the internet never forgets".

How should this field work?
---

If the field is set to 0, the note should be stored indefinitely.
If the field is set to a number higher than 0, the server should schedule a deletion of the note in the number of seconds that were provide as number by the TTL field.
If the field is set to a number lower than 0 the server should reject the note update/creation.

Note for 2.0 implementation:
---

The field should mainly be used to reserve this feature for 2.1, as I don't want to increate the scope of 2.0 any further. I'll leave it to the backend team to decide wheter the field is will be functional in 2.0. I would recommend to only accept notes with a TTL of 0 for now.

**Comment by DerMolly** [comment]
Should this maybe be saved as a concrete date in the database? Otherwise we would need to persist TTL deletion to survive restarts and such things…

**Comment by ErikMichelson** [comment]
Either persist it as a Date (probably a unix timestamp is simplest solution) or calculate the expiry based on creation-date and TTL on the fly when the delete-expired-notes-cronjob runs.

**Comment by DerMolly** [comment]
good point we have the creation date so we can calculate with that and the TTL. That's probably better

**Comment by ccoenen** [comment]
storing unix timestamp is not well readable.
storing an offset needs a calculation every time this is checked (and we have no benefit, because nobody wants to know "give me all notes with a ttl of 1440 seconds")

I would suggest storing a `DataTypes.DATE` - nothing fancy about it. We do this in various places already, and internally it does not take significantly more storage space. This field could be called `autoDeleteAfter` or something descriptive.

Another note: we already have a field that's at least similar. `deletedAt`. Maybe those should be combined?

**Comment by SISheogorath** [comment]
> good point we have the creation date so we can calculate with that and the TTL.

I don't think it would be a good idea to use the creation date for the TTL. The TTL should always be a relative time in seconds from now (i.e. the request arriving at the API). How it's internally stored is up to the implementation. This would also allow easy updates of the TTL (given one has the permissions for it) down the line.

**Comment by davidmehren** [comment]
I'm confused how this TTL should work. Does the TTL specify the lifetime after *note creation* or after the *last access* or after *setting the TTL*?
